### PR TITLE
CoreCLR compatibility

### PR DESCRIPTION
**Pull request by lstratman**
This is adds the ability for Edge.js to use the CoreCLR runtime instead of native or Mono.

### CODE REVIEW

File: src/edge-cs-coreclr/global.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "sdk": {
+        "version": "1.0.0-beta7-12331",
```
Comment (tjanczuk) [inline src/edge-cs-coreclr/global.json]: Should this be updated to RC1 or it does not matter?

File: src/edge-cs-coreclr/global.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+        "runtime": "coreclr",
+        "architecture": "x64"
```
Comment (tjanczuk) [inline src/edge-cs-coreclr/global.json]: What is the implication of the architecture choice here? Is this going to work with x86 edge after compilation (eventually, after that CLR bug is fixed).

File: src/edge-cs-coreclr/global.json (hunk 1 above)
Comment (lstratman) [inline src/edge-cs-coreclr/global.json]: I'm honestly not sure what this section of the global.json is even used for, it certainly doesn't have any effect on the compilation.  I think it's used by Visual Studio when it does package restores and Intellisense and needs to know which version of the framework to use.  That being said, I've gone ahead and updated the version for consistency's sake.

File: src/edge-cs-coreclr/global.json (hunk 2 above)
Comment (lstratman) [inline src/edge-cs-coreclr/global.json]: Per my previous comment, this should work just fine on x86.

**Comment by tjanczuk** [comment]
Thanks!

**Pull request merged**